Makefile

```diff
@@ -62,13 +131,22 @@ all: $(PROGRAMS) $(LIBRARIES)
 
 DEPDIR = .deps
 df = $(DEPDIR)/$(*D)/$(*F)
+
+ifeq ($(PE_ENV), GNU)
 %.o: %.cpp
 	@$(MKDIR_P) $(DEPDIR)/$(*D)
 	$(CPU_COMPILE) -MD -o $@ $<
 	@cp $*.d $(df).P; \
 	sed -e 's/#.*//' -e 's/^[^:]*: *//' -e 's/ *\\$$//' \
 		-e '/^$$/ d' -e 's/$$/ :/' < $*.d >> $(df).P; \
 	rm -f $*.d
```

Comment:
This should be supported by both PGI and Intel (at least) with some small adaptations perhaps. The whole point of this is let the compiler generate the `Makefile` rules instead of us writing them down and omitting some dependencies (thus messing perhaps parallel `make`).